models/line-item/src/transformers.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  LocalizedString,
+  TReferenceGraphql,
+} from '@commercetools-test-data/commons';
+import { Reference } from '@commercetools-test-data/commons';
+import { Transformer } from '@commercetools-test-data/core';
+import type { TLineItem, TLineItemGraphql, TLineItemRest } from './types';
+
+const transformers = {
+  default: Transformer<TLineItem, TLineItem>('default', {
+    buildFields: [
+      'name',
+      'productSlug',
+      'productType',
+      'variant',
+      'price',
+      'totalPrice',
+      'taxRate',
+      'supplyChannel',
+      'distributionChannel',
+    ],
+  }),
+  rest: Transformer<TLineItem, TLineItemRest>('rest', {
+    buildFields: [
+      'name',
+      'productSlug',
+      'productType',
+      'variant',
+      'price',
+      'totalPrice',
+      'taxRate',
+      'supplyChannel',
+      'distributionChannel',
+    ],
+    replaceFields: ({ fields }) => ({
+      ...fields,
+      supplyChannel: {
+        typeId: 'channel',
+        id: fields.supplyChannel.id,
+        key: fields.supplyChannel.key,
+      },
+      distributionChannel: {
+        typeId: 'channel',
+        id: fields.supplyChannel.id,
+        key: fields.supplyChannel.key,
+      },
+    }),
+  }),
+  graphql: Transformer<TLineItem, TLineItemGraphql>('graphql', {
+    buildFields: [
+      'name',
+      'productSlug',
+      'productType',
+      'variant',
+      'price',
+      'totalPrice',
+      'taxRate',
+      'supplyChannel',
+      'distributionChannel',
+    ],
+    addFields: ({ fields }) => {
+      const nameAllLocales = LocalizedString.toLocalizedField(fields.name);
+      const productSlugAllLocales = LocalizedString.toLocalizedField(
+        fields.productSlug
+      );
+      const supplyChannelRef: TReferenceGraphql = Reference.random()
+        .id(fields.supplyChannel.id)
+        .typeId('channel')
+        .buildGraphql();
+
+      const distributionChannelRef: TReferenceGraphql = Reference.random()
+        .id(fields.distributionChannel.id)
+        .typeId('channel')
+        .buildGraphql();
+
+      const productTypeRef: TReferenceGraphql = Reference.random()
+        .id(fields.distributionChannel.id)
+        .typeId('product-type')
+        .buildGraphql();
```

Comment:
Should these be built from the corresponding reference type fields?

Reply:
They basically are via setting the field ID, I opted for this because I was creating vanilla objects and manually adding a `__typename` when we have the `.buildGrahpql()` method to do just that

Reply:
You're absolutely right, apparently I was loopy by the time I got here.